generic/tests/kdump.py

```diff
@@ -215,16 +222,24 @@ def check_vmcore(test, vm, session, timeout):
 
     error_context.context("Waiting for kernel crash dump to complete",
                           logging.info)
-    session = vm.wait_for_login(timeout=timeout)
+    if vm.params.get("kdump_method") != "ssh":
+        session = vm.wait_for_login(timeout=timeout)
 
     error_context.context("Probing vmcore file...", logging.info)
     try:
         if vm.params.get("kdump_method") == "ssh":
             logging.info("Checking vmcore file on host")
-            process.run(vmcore_chk_cmd, shell=True)
+            utils_misc.wait_for(lambda:
+                                process.system(vmcore_chk_cmd,
+                                               shell=True) == 0,
+                                ignore_errors=True,
+                                timeout=200)
         else:
             logging.info("Checking vmcore file on guest")
-            session.cmd(vmcore_chk_cmd)
+            utils_misc.wait_for(lambda:
+                                session.cmd_status(vmcore_chk_cmd) == 0,
+                                ignore_errors=True,
+                                timeout=200)
```

Comment:
Would you help introduce what is this modification for?

Reply:
No matter `session.cmd(vmcore_chk_cmd)` or `utils_misc.wait_for()` they only return False/1 but won't raise Exception when vmcore file was not found, ritght?  So does it really catch any failure with this implementation?

Reply:
Sure, if we not add `wait_for`,the test maybe failed with cannot find vmcore file,but the root cause is vmcore file not generate completed,so I modify the code to wait the vmcore file generate.

If we not add `ignore_errors=True`,we can still catch exception,but the function "wait_for" will break with the exception, it seems not our expected result.Do you have any suggestions for this?

Reply:
I suppose the correct approach to check the vmcore file might be
```
if not utils_misc.wait_for(lambda: session.cmd_status(vmcore_chk_cmd) == 0, ignore_errors=True, timeout=200)
    test.fail(...)
```
Since utils_misc.wait_for() will return None or False if `ls` command does not return `0`.
No need to `try` then.

Reply:
Thank you for your guidance,updated the code.

Reply:
Thank you for your guidance,updated the code.